src/vs/workbench/electron-browser/workbench.ts

```diff
@@ -1142,6 +1142,24 @@ export class Workbench implements IPartService {
 		}
 	}
 
+	// Resize requested part along the main axis
+	// layout will do all the math for us and adjusts the other Parts
+	public resizePart(part: Parts, sizeChangePx: number, skipLayout?: boolean): TPromise<void> {
```

Comment:
I have to delve into the promises use better
I guess it is just the layout function that needs the promise structure